app/assets/javascripts/discourse/app/controllers/share-topic.js

```diff
@@ -32,6 +33,17 @@ export default Controller.extend(
       if (this.model && this.model.read_restricted) {
         this.restrictedGroupWarning();
       }
+
+      scheduleOnce("afterRender", this, this.selectUrl);
+    },
+
+    selectUrl() {
+      const input = document.querySelector("input.invite-link");
```

Comment:
You should check if the object is destroyed/isDestroying here, in case the scheduled event fires after it is torn down. (This is more likely in tests, but still good to catch.)

Reply:
I think the destroyed/isDestryoing check can be added only to components, right? I changed the `input` below to skip the selection range logic if the input does not exist.

Reply:
> I think the destroyed/isDestryoing check can be added only to components, right?

Nope! `isDestroying` and `isDestroyed` are properties of pretty much all ember objects: components, controller, routes, router, application instance, etc. 🙂